chart.js.d.ts

```diff
@@ -0,0 +1,605 @@
+  function Bar(context: CanvasRenderingContext2D, args: {data: BarChartData, options?: BarChartOptions}): BarChartInstance;
+
+  function Radar(context: CanvasRenderingContext2D, args: {data: RadarChartData, options?: RadarChartOptions}): RadarChartInstance;
+
+  function PolarArea(context: CanvasRenderingContext2D, args: {data: PolarAreaChartData, options?: PolarAreaChartOptions}): PolarAreaChartInstance;
+
+  function Bubble(context: CanvasRenderingContext2D, args: {data: CircularChartData, options?: CircularChartOptions}): CircularChartInstance;
+
+  function Doughnut(context: CanvasRenderingContext2D, args: {data: CircularChartData, options?: CircularChartOptions}): CircularChartInstance;
+
+  const defaults: {
+    global: ChartOptions
+    line: LineChartOptions
+    bar: BarChartOptions
+    radar: RadarChartOptions
+    polarArea: PolarAreaChartOptions
+    doughnut: CircularChartOptions
+    pie: CircularChartOptions
+  };
+
+  // Check required, what fields does the defaultConfig object contain?
+  const scaleService: {
+    registerScaleType<T extends Scale>(key: string, scale: T, defaultConfig: Object): void
+  };
+
+  interface ChartData {
+    labels: Array<string>
+    datasets?: Array<ChartDataSet>
+  }
+
+  interface ChartDataSet {
+    data: Array<number>
+  }
+
+  interface ChartOptions {
+    responsive?: boolean
+    responsiveAnimationDuration?: number
+    maintainAspectRatio?: boolean
+    events?: Array<string>
+    hover?: {
+      onHover?: (activeElements: Array<Element>) => void // Check required, what fields does an element contain? element interface?
+      mode?: string
+      animationDuration?: number
+    }
+    onClick?: (activeElements: Array<Element>) => void // Check required
+    defaultColor?: string
+    defaultFontColor?: string
+    defaultFontFamily?: string
+    defaultFontSize?: number
+    defaultFontStyle?: string
+    legendCallback?: (chart: ChartInstance) => string // Check required
+    title?: ChartTitleOptions
+    legend?: ChartLegendOptions
+    tooltips?: ChartTooltipsOptions
+    animation?: ChartAnimationOptions
+    elements?: ChartElementsOptions
+  }
+
+  interface ChartTitleOptions {
+    display?: boolean
+    position?: string
+    fullWidth?: boolean
+    fontColor?: string
+    fontFamily?: string
+    fontSize?: number
+    fontStyle?: string
+    padding?: number
+    text?: string
+  }
+
+  interface ChartLegendOptions {
+    display?: boolean
+    position?: string
+    fullWidth?: boolean
+    onClick?: (event: Event, legendItem: LegendItem) => void // Check required, is Event the correct type here?
+    labels?: {
+      labelsboxWidth?: number
+      labelsfontSize?: number
+      labelsfontStyle?: string
+      labelsfontColor?: string
+      labelsfontFamily?: string
+      labelspadding?: number
+      labelsgenerateLabels?: (data: ChartData) => Array<LegendItem> // Check required
+    }
+  }
+
+  // Check required, which fields are optional?
+  interface LegendItem {
```

Comment:
I like calling this out as a top level interface. I feel we should add this to the docs explicitly.